addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,64 @@
+        return [(0, 0, {
+            'factor_percent': line.factor_percent,
+            'repartition_type': line.repartition_type,
+            'account_id': line.repartition_type == 'tax' and line.account_id.id or None,
+            'company_id': line.company_id.id,
+            'sequence': line.sequence,
+        }) for line in original_rep_lines]
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_group_id = self.env.ref('l10n_eu_service.tax_group_oss').id
+        for company in self:
+            tax_ids = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id)])
+            for domestic_tax in tax_ids:
+                for country in eu_countries - company.country_id:
+                    tax_amount = EU_TAX_MAP.get(company.country_id.code + str(domestic_tax.amount) + country.code, False)
+                    if tax_amount:
+                        tax_name =_('%(rate)s%% %(country)s %(label)s', rate=tax_amount, country=country.code, label=country.vat_label)
+                        foreign_tax = self.env['account.tax'].search([('name', '=', tax_name), ('company_id', '=', company.id)])
+                        if not foreign_tax:
+                            foreign_tax = self.env['account.tax'].create({
+                                'name': tax_name,
+                                'amount': tax_amount,
+                                'invoice_repartition_line_ids': self._get_repartition_line_copy_values(domestic_tax.invoice_repartition_line_ids),
+                                'refund_repartition_line_ids': self._get_repartition_line_copy_values(domestic_tax.refund_repartition_line_ids),
+                                'type_tax_use': 'sale',
+                                'description': "%s%%" % tax_amount,
+                                'tax_group_id': tax_group_id,
+                                'sequence': 1000,
+                                'company_id': company.id,
+                            })
+                        fpos = self.env['account.fiscal.position'].search([
+                            ('country_id', '=', country.id),
+                            ('company_id', '=', company.id),
+                            ('auto_apply', '=', True),
+                            ('vat_required', '=', False)], limit=1)
+                        if not fpos:
+                            new_fpos_name = _('OSS B2C %s', country.name)
+                            fpos = self.env['account.fiscal.position'].create({
+                                'name': new_fpos_name,
+                                'company_id': company.id,
+                                'vat_required': False,
+                                'auto_apply': True,
+                                'country_id': country.id,
+                                'account_ids': False,
```

Comment:
No need to specify account_ids if it's just empty.

Reply:
removed. If we adapt the wizard to use this method instead, we may want to bring it back.